tests/Module/WeightCalculatorTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare( strict_types=1 );
+
+namespace packeta\tests\Module;
+
+use Packetery\Module\Framework\WcAdapter;
+use Packetery\Module\Options\OptionsProvider;
+use Packetery\Module\WeightCalculator;
+use PHPUnit\Framework\MockObject\Exception;
+use PHPUnit\Framework\TestCase;
+use WC_Order;
+use WC_Order_Item_Product;
+use WC_Product;
+
+class WeightCalculatorTest extends TestCase {
+	private function createOrderItemMock( float $productQuantity, float $weight ): WC_Order_Item_Product {
+		$wcProduct = $this->createMock( WC_Product::class );
+		$wcProduct
+			->method( 'get_weight' )
+			->willReturn( $weight );
+
+		$wcOrderItemProduct = $this->createMock( WC_Order_Item_Product::class );
+		$wcOrderItemProduct
+			->method( 'get_product' )
+			->willReturn( $wcProduct );
+		$wcOrderItemProduct
+			->method( 'get_quantity' )
+			->willReturn( $productQuantity );
+
+		return $wcOrderItemProduct;
+	}
+
+	/**
+	 * @return array<array{0: bool, 1: float, 2: float, 3: float, 4: array[]}>
+	 */
+	public static function getOrderWeightCalculationProvider(): array {
+		return [
```

Comment:
U data providerů by měly být pojmenované klíče, aby bylo jasné, co se testuje a co je to za hodnoty.